Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/QrCodeScannerViewModel.kt

```diff
@@ -35,15 +37,20 @@ class QrCodeScannerViewModel @AssistedInject constructor(
     fun onImportFile(fileUri: Uri) = launch {
         result.postValue(InProgress)
         Timber.tag(TAG).d("onImportFile(fileUri=$fileUri)")
-        when (val parseResult = qrCodeFileParser.decodeQrCodeFile(fileUri)) {
-            is QrCodeFileParser.ParseResult.Failure -> {
-                Timber.tag(TAG).d(parseResult.exception, "parseResult failed")
-                result.postValue(Error(error = parseResult.exception))
-            }
-            is QrCodeFileParser.ParseResult.Success -> {
-                Timber.tag(TAG).d("parseResult=$parseResult")
-                onScanResult(parseResult.text)
+        try {
+            when (val parseResult = qrCodeFileParser.decodeQrCodeFile(fileUri)) {
+                is QrCodeFileParser.ParseResult.Failure -> {
+                    Timber.tag(TAG).d(parseResult.exception, "parseResult failed")
+                    result.postValue(Error(error = parseResult.exception))
+                }
+                is QrCodeFileParser.ParseResult.Success -> {
+                    Timber.tag(TAG).d("parseResult=$parseResult")
+                    onScanResult(parseResult.text)
+                }
```

Comment:
we have to do the same in Submission On behalf flow `OrganizerWarnQrCodeScannerViewModel`

Reply:
`OrganizerWarnQrCodeScannerViewModel` is update